### DISCUSSION TITLE
Is it possible that SQuAD has some errors?

### DISCUSSION THREAD

**Discussion by ggoggam** [seominjoon/kaist-ai605#27] [Q&A] [answered]
I found in one of the examples, the `answer_start` did not match the character index in the `context`.

```
{
     "context":  "Kazakhstan: The first torchbearer in Almaty, where the Olympic torch arrived for the first time ever on April 2, was the President of Kazakhstan Nursultan Nazarbaev. The route ran 20 km from Medeo stadium to Astana Square. There were reports that Uighur activists were arrested and some were deported back to China.", 
     "answers": {
           "answer_start": [182], 'text': ['20']
     },
    "id": "56db0495e7c41114004b4c52",
    "question": "What was the distance in kilometers for the route in Kazakhstan?", 
    "title": "2008_Summer_Olympics_torch_relay"
}
```

In this example, the character index of answer `"20"` begins at 180, yet `answer_start` says otherwise. Are we supposed to fix these cases? I am not sure how I should be handling these.

EDIT: It seems that there are some cases where character indices do not match, so one must use some tricks to match the token index.

**Comment by seominjoon** [comment, marked as answer]
This is something I did not know about. It's a small portion so shouldn't be a big deal at the end, but will let you know after I investigate into it.

**Reply by ggoggam** [reply]
It seems there are some cases where character indices are messed up, but I think you can kind of bypass it by checking if your regex match span and character index span (obtained from dataset) have some overlap.
